controllers/stargate/stargate_controller.go

```diff
@@ -265,6 +265,19 @@ func (r *StargateReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		}
 	}
 
+	// create Vector configmap
+	if vectorReconcileResult, err := r.reconcileVector(ctx, *stargate, actualDc, r.Client, logger); err != nil {
+		return vectorReconcileResult, err
+	} else if vectorReconcileResult.Requeue {
+		return vectorReconcileResult, nil
+	}
+
+	// inject Vector
+	if err := InjectVectorAgentForStargate(stargate, desiredDeployments, actualDc.Name, actualDc.Spec.ClusterName, logger); err != nil {
```

Comment:
As stated in another comment, this function doesn't return any error, so we could remove the error handling code here.